src/backend/distributed/transaction/backend_data.c

```diff
@@ -947,6 +950,50 @@ citus_backend_gpid(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * get_nodeid_for_gpid returns node id for the global process with given global pid
+ */
+Datum
+get_nodeid_for_gpid(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	uint64 globalPID = PG_GETARG_INT64(0);
+
+	bool missingOk = false;
+	PG_RETURN_INT32(ExtractNodeIdFromGlobalPID(globalPID, missingOk));
+}
+
+
+/*
+ * get_pid_for_gpid returns process id for the global process with given global pid
+ */
+Datum
+get_pid_for_gpid(PG_FUNCTION_ARGS)
+{
+	CheckCitusVersion(ERROR);
+
+	uint64 globalPID = PG_GETARG_INT64(0);
+
+	PG_RETURN_INT32(ExtractProcessIdFromGlobalPID(globalPID));
+}
+
+
+/*
+ * assign_global_pid is the wrapper UDF for AssignGlobalPID and is only meant for use
+ * in tests.
+ */
+Datum
+assign_global_pid(PG_FUNCTION_ARGS)
```

Comment:
Hmm, I'm not very sure if we can gracefully handle changing the GPID normally. 

I'd at least suggest that we move this UDF to src/backend/distributed/test folder and never expose in the backend_data.

Also, name it like `test_assign_global_pid` or such.